src/django_components/slots.py

```diff
@@ -335,23 +351,88 @@ def render(self, context: Context, name: str, **kwargs: Any) -> SafeString:
             # by specifying the fill both by explicit slot name and implicitly as 'default'.
             if (
                 slot_name != DEFAULT_SLOT_KEY
-                and component_ctx.fills.get(slot_name, False)
-                and component_ctx.fills.get(DEFAULT_SLOT_KEY, False)
+                and slot_fills.get(slot_name, False)
+                and slot_fills.get(DEFAULT_SLOT_KEY, False)
             ):
                 raise TemplateSyntaxError(
-                    f"Slot '{slot_name}' of component '{component_ctx.component_name}' was filled twice: "
+                    f"Slot '{slot_name}' of component '{component_name}' was filled twice: "
                     "once explicitly and once implicitly as 'default'."
                 )
 
         # If slot is marked as 'default', we use the name 'default' for the fill,
         # IF SUCH FILL EXISTS. Otherwise, we use the slot's name.
-        if is_default and DEFAULT_SLOT_KEY in component_ctx.fills:
+        if is_default and DEFAULT_SLOT_KEY in slot_fills:
             fill_name = DEFAULT_SLOT_KEY
         else:
             fill_name = slot_name
 
-        if fill_name in component_ctx.fills:
-            slot_fill_fn = component_ctx.fills[fill_name]
+        # NOTE: TBH not sure why this happens. But there's an edge case when:
```

Comment:
This section right here is what took me so long to debug. And after all this time, I'm still not sure why things happen this way. But at least things are working.